Sources/EngineKit/HashableExtensions.swift

```diff
@@ -46,118 +52,173 @@ extension Instructions: Hashable {
 			hasher.combine(args)
+
 		case let .callMethod(address, methodName, args):
 			hasher.combine("callMethod")
 			hasher.combine(address)
 			hasher.combine(methodName)
 			hasher.combine(args)
+
 		case let .callRoyaltyMethod(address, methodName, args):
 			hasher.combine("callRoyaltyMethod")
 			hasher.combine(address)
 			hasher.combine(methodName)
 			hasher.combine(args)
+
 		case let .callMetadataMethod(address, methodName, args):
 			hasher.combine("callMetadataMethod")
 			hasher.combine(address)
 			hasher.combine(methodName)
 			hasher.combine(args)
+
 		case let .assertWorktopContainsAny(resourceAddress):
 			hasher.combine("assertWorktopContainsAny")
 			hasher.combine(resourceAddress)
+
 		case let .callDirectVaultMethod(address, methodName, args):
 			hasher.combine("callDirectVaultMethod")
 			hasher.combine(address)
 			hasher.combine(methodName)
 			hasher.combine(args)
+
 		case let .allocateGlobalAddress(packageAddress, blueprintName):
 			hasher.combine("allocateGlobalAddress")
 			hasher.combine(packageAddress)
 			hasher.combine(blueprintName)
+
 		case .dropNamedProofs:
+			mustNotHaveAssociatedValue(.dropNamedProofs)
 			hasher.combine("dropNamedProofs")
+
 		case .dropAuthZoneProofs:
+			mustNotHaveAssociatedValue(.dropAuthZoneProofs)
 			hasher.combine("dropAuthZoneProofs")
+
 		case .dropAuthZoneRegularProofs:
+			mustNotHaveAssociatedValue(.dropAuthZoneRegularProofs)
 			hasher.combine("dropAuthZoneRegularProofs")
+
 		case .dropAuthZoneSignatureProofs:
+			mustNotHaveAssociatedValue(.dropAuthZoneSignatureProofs)
 			hasher.combine("dropAuthZoneSignatureProofs")
+
 		case let .callRoleAssignmentMethod(address, methodName, args):
 			hasher.combine("callRoleAssignmentMethod")
 			hasher.combine(address)
 			hasher.combine(methodName)
 			hasher.combine(args)
 		}
 	}
+}
 
-	public static func == (lhs: Instruction, rhs: Instruction) -> Bool {
-		switch (lhs, rhs) {
+// MARK: - Instruction + Equatable
+extension Instruction: Equatable {
+	public static func == (lhsOuter: Instruction, rhsOuter: Instruction) -> Bool {
+		switch lhsOuter {
+		case .takeAllFromWorktop, .takeFromWorktop, .takeNonFungiblesFromWorktop, .returnToWorktop, .assertWorktopContains, .assertWorktopContainsNonFungibles, .popFromAuthZone, .dropAuthZoneProofs, .dropAuthZoneRegularProofs, .dropAuthZoneSignatureProofs, .dropAllProofs, .dropNamedProofs, .pushToAuthZone, .createProofFromAuthZoneOfAmount, .createProofFromAuthZoneOfNonFungibles, .createProofFromAuthZoneOfAll, .burnResource, .cloneProof, .dropProof, .assertWorktopContainsAny, .createProofFromBucketOfAmount, .createProofFromBucketOfNonFungibles, .createProofFromBucketOfAll, .callFunction, .callMethod, .callRoyaltyMethod, .callMetadataMethod, .callRoleAssignmentMethod, .allocateGlobalAddress, .callDirectVaultMethod:
+			break
+			// If switch does not compile, you MUST handle it below, in EQ impl.
+		}
+		func `true`(mustNotHaveAssociatedValue _: Instruction) -> Bool {
```

Comment:
We should not have functions named "true"

Reply:
will rename to `equal`

Reply:
Why not just remove it and return `true`?